src/content/components/StyleDef.js

```diff
@@ -0,0 +1,46 @@
+// inspired from https://gist.github.com/jviereck/9a71734afcfe848ddbe2 -- simplified
+import React, { PureComponent, PropTypes } from 'react';
+
+export class StyleDef extends PureComponent {
```

Comment:
In short, it's mainly because CSS has `{` everywhere and so it's painful to manipulate directly with JSX.

The original version was much more complex (and buggy) because it was keeping track of whether the content was already added, doing reference counting and such. I removed this because I don't need it here.

That said I admit I haven't tried inserting a `<style>` directly at the root, but I wouldn't really like this.